ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java

```diff
@@ -123,6 +128,26 @@ public ForkManifest getForkManifest() {
     return forkManifest;
   }
 
+  public Collection<ForkAndMilestone> getEnabledMilestones() {
+    return forkManifest.getForkSchedule().stream()
+        .map(fork -> new ForkAndMilestone(fork, getMilestoneForFork(fork)))
+        .collect(toList());
+  }
+
+  private Milestone getMilestoneForFork(final Fork fork) {
+    final SpecConfig specConfig = getSpecConfig(fork.getEpoch());
```

Comment:
Currently, the code is structured so that we only ever have a single `SpecConfig` instance - it isn't really versioned like everything else, it's just a different subtype depending on the loaded config.  So, we should be able to inject the single `SpecConfig` instance into `ForkManifest` so that we can move this logic there.

Reply:
It feels weird to me that everywhere else is getting the `SpecConfig` based off of an epoch and then this is the one place we break that concept.  Its true that the plan is to never change constants so we could depend on that but I don't like having an inconsistent approach to it.

And after some time away from it, I think I'm actually happier with it in `Spec` - the code that actually calls is has the `spec` so moving it into `ForkManager` just means an extra get call (ie `spec.getForkManager().getEnabledMilestones()`.  The bigger thing though is that current `ForkManifest` is very tightly focussed on tracking the `Fork` list and that's it.  I kind of like that focus rather than having it know about the spec config and the milestone which is really a way to identify the specific logic used in a fork.

So going to merge it with this in `Spec` but I don't feel strongly about it so happy to move it in a follow up if you'd prefer.

Reply:
No strong opinion - sounds like a good plan.  I was thinking we should go back in and clean up the handling of `SpecConfig`, but haven't really looked into it.  And maybe it's not so bad to keep accessing it through `SpecVersion` in case we wanted to change the config management in the future ...